packages/ra-core/src/form/useInput.ts

```diff
@@ -88,7 +88,10 @@ export const useInput = <ValueType = any>(
         rules: {
             validate: async (value, values) => {
                 if (!sanitizedValidate) return true;
-                const error = await sanitizedValidate(value, values, props);
+                const error = await sanitizedValidate(value, values, {
+                    ...props,
+                    source: finalSource,
+                });
```

Comment:
**Open question:** I wonder if it's best to _override_ the `source` prop, or to pass an additional `finalSource` prop. The [docs](https://marmelab.com/react-admin/Validation.html#per-input-validation-custom-function-validator) simply say that the signature should be `(value, allValues, props)`, so technically we could choose either solution.
What do you guys think?

Reply:
I'm +1 for passing both `source` and `finalSource`

Reply:
done